### ISSUE TITLE
Warning is fatal and stops further rule checks

### ISSUE THREAD

**Issue by nkakouros** [ansible/ansible-lint#2868]
##### Summary

A `warning[empty-playbook]` warning is fatal, when it should be just that, a warning.

##### Issue Type

- Bug Report

##### OS / ENVIRONMENT

```console (paste below)
ansible-lint --version
ansible-lint 6.10.2 using ansible 2.14.1
```

- ansible installation method: pip
- ansible-lint installation method: pip

##### STEPS TO REPRODUCE

I have a simple role, with 2 files:

```
# myrole/tasks/main.yml
- debug: ~
```

```
myrole/molecule/default/molecule.yml
---
```

The molecule file is empty because I use a global one for all my roles.

##### Desired Behavior

The empty playbook warning should just be listed as such and I should receive an error about the unnamed tasks in tasks/main.yml and other errors.

##### Actual Behavior

```paste below
(.venv) [nikos@instance-2 myrole]$ ansible-lint molecule/default/moelcule.yml tasks/main.yml
WARNING  Listing 1 violation(s) that are fatal
warning[empty-playbook]: Empty playbook, nothing to do (warning)
molecule/default/molecule.yml:1:1

You can skip specific rules or tags by adding them to your configuration file:
# .config/ansible-lint.yml
warn_list:  # or 'skip_list' to silence them completely
  - experimental  # all rules tagged as experimental

                       Rule Violation Summary
 count tag                     profile rule associated tags
     1 warning[empty-playbook]         core, experimental (warning)

Passed with production profile, 5/5 star rating: 0 failure(s), 1 warning(s) on 2 files.
```

The empty playbook is a warning. I don't get why it is fatal! 

<del>The only way to make ansible-lint go further is to add `warning[empty-playbook]` to `skip_list`. </del>
There is not way to ignore this fatal error.

I don't know why molecule.yml is treated as a playbook and I would be happy to learn how to instruct ansible-lint to not do that, but regardless of this, it seems wrong that a warning could be fatal.

**Comment by isuftin** [comment]
.ansible-lint

```yaml
exclude_paths:
  - ./molecule
  - ./whatever_molecule_playbooks_to_ignore.yml
```

**Comment by nkakouros** [comment]
But I still want molecule.yml to be included in any further ansible-lint checks (as well as yamllint through ansible-lint). The important part of my issue, though, is why warning is fatal.

**Comment by nkakouros** [comment]
The issue is that an empty playbook is treated as an error by `ansible-playbook --syntac-check` and ansible-lint uses the output from that command to stop processing if there are syntax errors:

https://github.com/ansible/ansible-lint/blob/2f688cc04034208c461620bb1e08abf705978039/src/ansiblelint/runner.py#L138-L163

This is why adding `warning` to `skip_list` has no effect.

The docs also contradict each other:

- [syntax-check](https://ansible-lint.readthedocs.io/rules/syntax-check/) says: 
  > Our linter runs ansible-playbook --syntax-check on all playbooks, and if any of these reports a syntax error, this stops any further processing of these files.
  >
  > This error cannot be disabled due to being a prerequisite for other steps. You can exclude these files from linting, but it is better to make sure they can be loaded by Ansible.
  This obviously covers `empty-playbook`.
- [warning](https://ansible-lint.readthedocs.io/rules/warning/) says:
  > warning is a special type of internal rule that is used to report generic runtime warnings found during execution. As stated by its name, they are not counted as errors, so they do not influence the final outcome.
  when apparently `empty-playbook` does influence the final outcome.

There are two solutions here:

1. Either `warning[empty-playbook]` should not be a [`WarningRule`](https://github.com/ansible/ansible-lint/blob/2f688cc04034208c461620bb1e08abf705978039/src/ansiblelint/rules/syntax_check.py#L140-L141) and is left as an `AnsibleSyntaxCheckRule` and the docs are updated to remove `empty-playbook` from [the warning page](https://ansible-lint.readthedocs.io/rules/warning/),
2. Or `warning[empty-playbook]` is ignored in the [syntax check](https://github.com/ansible/ansible-lint/blob/2f688cc04034208c461620bb1e08abf705978039/src/ansiblelint/runner.py#L158):

  https://github.com/ansible/ansible-lint/blob/2f688cc04034208c461620bb1e08abf705978039/src/ansiblelint/runner.py#L158

  is changed to:

  ```python
  matches.extend([match for match in data if match.tag != 'warning[empty-playbook]'])
  ```

@ssbarnea Any thoughts about this?
